sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java

```diff
@@ -0,0 +1,1069 @@
+            data = BinaryData.fromBytes(Base64.getEncoder().encode(content.toBytes()));
+        }
+
+        SendEventMessage message = new SendEventMessage()
+            .setEvent(eventName)
+            .setData(data)
+            .setDataType(dataType.toString())
+            .setAckId(ackId);
+
+        Mono<Void> sendMessageMono = sendMessage(message);
+        Mono<WebPubSubResult> responseMono = options.isFireAndForget()
+            ? sendMessageMono.then(Mono.just(new WebPubSubResult(null, false)))
+            : sendMessageMono.then(waitForAckMessage(ackId));
+        return responseMono.retryWhen(sendMessageRetrySpec);
+    }
+
+    /**
+     * Receives group message events.
+     *
+     * @return the Publisher of group message events.
+     */
+    public Flux<GroupMessageEvent> receiveGroupMessageEvents() {
+        return groupMessageEventSink.asFlux();
+    }
+
+    /**
+     * Receives server message events.
+     *
+     * @return the Publisher of server message events.
+     */
+    public Flux<ServerMessageEvent> receiveServerMessageEvents() {
+        return serverMessageEventSink.asFlux();
+    }
+
+    /**
+     * Receives connected events.
+     *
+     * @return the Publisher of connected events.
+     */
+    public Flux<ConnectedEvent> receiveConnectedEvents() {
+        return connectedEventSink.asFlux();
+    }
+
+    /**
+     * Receives disconnected events.
+     *
+     * @return the Publisher of disconnected events.
+     */
+    public Flux<DisconnectedEvent> receiveDisconnectedEvents() {
+        return disconnectedEventSink.asFlux();
+    }
+
+    /**
+     * Receives stopped events.
+     *
+     * @return the Publisher of stopped events.
+     */
+    public Flux<StoppedEvent> receiveStoppedEvents() {
+        return stoppedEventSink.asFlux();
+    }
+
+    /**
+     * Receives re-join group failed events.
+     *
+     * @return the Publisher of re-join failed events.
+     */
+    public Flux<RejoinGroupFailedEvent> receiveRejoinGroupFailedEvents() {
+        return rejoinGroupFailedEventSink.asFlux();
+    }
+
+    private long nextAckId() {
+        return ACK_ID.getAndUpdate(value -> {
+            // keep positive
+            if (++value < 0) {
+                value = 0;
+            }
+            return value;
+        });
+    }
+
+    private Flux<AckMessage> receiveAckMessages() {
+        return ackMessageSink.asFlux();
+    }
+
+    private Mono<Void> sendMessage(WebPubSubMessage message) {
+        return checkStateBeforeSend().then(Mono.create(sink -> {
+            if (logger.canLogAtLevel(LogLevel.VERBOSE)) {
+                try {
+                    String json = JacksonAdapter.createDefaultSerializerAdapter()
+                        .serialize(message, SerializerEncoding.JSON);
+                    logger.atVerbose().addKeyValue("message", json).log("Send message");
+                } catch (IOException e) {
+                    //
```

Comment:
We need to handle this exception. Log and bubble up - `sink.error(new UncheckedIOExcpetion(e);`.

Reply:
This 2 is in VERBOSE logging.

At present, I commented out these logging.
1. Netty impl also logs TextFrame
2. `serialize` here has some penalty to performance